pytorch_lightning/strategies/deepspeed.py

```diff
@@ -399,7 +398,7 @@ def _setup_model_and_optimizers(self, model: Module, optimizers: List[Optimizer]
         return self.model, [optimizer]
 
     def _setup_model_and_optimizer(
-        self, model: Module, optimizer: Optimizer, lr_scheduler: Optional[_LRScheduler] = None
+        self, model: Module, optimizer: Optimizer, lr_scheduler: Optional[LRSchedulerTypeUnion] = None
```

Comment:
Does this typing mean LR scheduler customisation is not supported with DeepSpeed strategy? If so, should we note in the docs that only native PyTorch LR schedulers are supported in DeepSpeed?

Reply:
lightning handles the deepspeed scheduler: https://github.com/PyTorchLightning/pytorch-lightning/blob/98ea79b8b07fd30ed718d28052d1bd8a586d3796/pytorch_lightning/strategies/deepspeed.py#L481-L487

Reply:
Thanks for the reference. As the customisation is supported with DeepSpeed, isn't the type `LRSchedulerTypeUnion` too tight? I might be missing something, but it should probably be something looser such as `_SupportsStateDict` instead of `LRSchedulerTypeUnion`, no? Same applies for the type of `LightningModule.lr_scheduler_step`'s `scheduler`.

https://github.com/PyTorchLightning/pytorch-lightning/blob/65bda5f4f1f0e898cb710ea762a7c89eb81dcb8f/pytorch_lightning/utilities/types.py#L96

Reply:
these types are used for LightningCLI to generate docs for `help`. If you notice, there are separate types defined for _LRScheduler and ReduceLROnPlateau but they are not used here for the same reason. I asked Carlos, he suggested leaving it for now and in case this comes up for a CLI request, we will update this later.

Reply:
Ah, that totally makes sense. Thanks for the clarification!